File: API/DTOs/Posts/PostSummary.cs
using API.DTOs.Topics;
using API.Entities;

namespace API.DTOs.Posts;

public class PostSummary
{
    public Guid Id { get; init; }
    public required string Title { get; init; }
    public required string Description { get; init; }
    public List<TopicDto> Topics { get; set; }
    public DateTime CreatedOn { get; init; }
    public string ThumbnailUrl { get; set; }
}

File: client/src/Types/PostSummary.ts
import { TopicColor } from './TopicColor';

export interface TopicDto {
  id: string;       // adjust type depending on your C# TopicDto definition
  name: string;
  color: TopicColor;
}

export interface PostSummaryDto {
  id: string;                      // Guid → string
  title: string;
  description: string;
  topics: TopicDto[];
  createdOn: string;               // DateTime → ISO string in JSON
  thumbnail?: string | null;    // nullable in C#
  status: 'Draft' | 'Published' | 'Deleted';
}
